tests/integration/ef_tests/utils.py

```diff
@@ -0,0 +1,62 @@
+import json
+import os
+
+from tests.utils.uint256 import uint256_to_int
+
+# Root of the GeneralStateTest in BlockchainTest format
+EF_GENERAL_STATE_TEST_ROOT_PATH = (
+    "./tests/integration/ef_tests/test_data/BlockchainTests/GeneralStateTests/"
+)
+
+DEFAULT_NETWORK = "Shanghai"
+
+
+def display_storage(uint256_tuple):
+    return hex(uint256_to_int(*uint256_tuple))
+
+
+def is_account_eoa(state: dict) -> bool:
+    return state.get("code") in [None, "0x"] and not state.get("storage")
+
+
+def process_json_file(filepath):
+    with open(filepath, "r") as f:
+        json_content = json.load(f)
+    return [
+        (test_name, test_content)
+        for test_name, test_content in json_content.items()
+        if test_content["network"] == DEFAULT_NETWORK
+    ]
+
+
+def load_ef_blockchain_tests_from_path(directory_path):
+    ef_tests = []
+    if os.path.isfile(directory_path):  # Handle case where path is a file
+        ef_tests.extend(process_json_file(directory_path))
+    else:  # Handle case where path is a directory
+        for dirpath, _, filenames in os.walk(directory_path):
+            for filename in filenames:
+                if filename.endswith(".json"):
+                    filepath = os.path.join(dirpath, filename)
+                    ef_tests.extend(process_json_file(filepath))
+
+    return ef_tests
+
+
+def load_ef_blockchain_tests(relative_path):
+    """
+    Load Ethereum Foundation tests from the directory or file specified
+    under a fixed root path.
+    """
+    full_path = os.path.join(EF_GENERAL_STATE_TEST_ROOT_PATH, relative_path)
+    if not os.path.exists(full_path):
+        raise ValueError(f"The path {full_path} does not exist.")
+
+    return load_ef_blockchain_tests_from_path(full_path)
+
+
+def load_all_ef_blockchain_tests():
+    """
+    Load all ef blockchain tests from root directory, recursively. This allows users to pattern match with the -k (keyword) option.
+    """
+    return load_ef_blockchain_tests(".")
```

Comment:
why these 3 functions in drawer? (does it mean something in english?)

Reply:
this looks like something for a next PR left here indeed

Reply:
let's just remove all these and use

```
for dirpath, _, filenames in os.walk(directory_path):
    for filename in filenames:
        if filename.endswith(".json"):
            filepath = os.path.join(dirpath, filename)
            ef_tests.extend(process_json_file(filepath))
```

directly in the `conftest.py`

also standard way is is to name the arg `for (root, dirs, files) in os.walk` 

and use list comprehension instead as it makes it easier to get what is inside the list

```
    return [
        process_json_file(os.path.join(dirpath, filename))
        for dirpath, _, filenames in os.walk(directory_path)
        for filename in filenames
        if filename.endswith(".json")
    ]
```

and even with the filtering directly here

```
return [
    {
        name: content
        for name, content in json.loads(
            (Path(dirpath / filename).read_text())
        ).item()
        if content["network"] == DEFAULT_NETWORK
    }
    for dirpath, _, filenames in os.walk(directory_path)
    for filename in filenames
    if filename.endswith(".json")
]
```

Reply:
now i see the pythonic light